scheds/rust/scx_rusty/src/bpf/main.bpf.c

```diff
@@ -606,6 +609,31 @@ s32 BPF_STRUCT_OPS(rusty_select_cpu, struct task_struct *p, s32 prev_cpu,
 		}
 	}
 
+	/*
+	 * The task might also be forked/execd so check for wake up only
+	 */
+	if ((wake_flags & SCX_WAKE_TTWU) && !fifo_sched) {
+		u32 dom_id = taskc->dom_id;
+		struct dom_ctx *domc;
+		u64 *cpu_vtime;
+
+		if (!(domc = bpf_map_lookup_elem(&dom_data, &dom_id))) {
+			scx_bpf_error("Failed to lookup dom[%u]", dom_id);
+			goto enoent;
+		}
+
+		if (vtime_before(p->scx.dsq_vtime, domc->vtime_now - slice_ns)) {
+			cpu_vtime = MEMBER_VPTR(current_cpu_vtime, [domc->vtime_now_cpu]);
+
+			if (cpu_vtime && *cpu_vtime == domc->vtime_now && bpf_cpumask_test_cpu(domc->vtime_now_cpu, p->cpus_ptr)) {
```

Comment:
What does `*cpu_time == domc->vtime_now` do?

Reply:
This "attempts" (though it's racy) to make sure we don't try to preempt the same CPU running the same task if we already did so (hence the setting of *cputime to 0 right below).

Reply:
Hmm.. unless it's known this does happen to a noticeable degree, I'd rather not have it. Trying to reduce the size of race window whose size is not known can become confusing while not adding much in practice.